### PR TITLE
Port test cases to Karma

### PR DESCRIPTION
**Pull request by Gusted**
- Port specific test cases to Karma and make them work.
- Add 'debug.log' to .gitignore as it will output a debug.log if chrome/firefox gets a internal error.

### CODE REVIEW

File: tests/inject/dynamic/style-override.tests.ts [hunk 1]
```diff
@@ -74,4 +74,91 @@ describe('Style override', () => {
...
+            '<style class="testcase-style"></style>',
+            '<h1>Some test foor...... <strong>Oh uhm a pink background</strong></h1>',
+        );
+        createOrUpdateDynamicTheme(theme, null, false);
+        const style: HTMLStyleElement = document.querySelector('.testcase-style');
+        style.sheet.insertRule('h1 { color: gray }');
+        style.sheet.insertRule('strong { color: red }');
+
+        (style as HTMLStyleElement).sheet.insertRule('html { background-color: pink }');
+        await timeout(1000);
```
Comment (alexanderby) [inline tests/inject/dynamic/style-override.tests.ts:140]: 1000 is too much, maybe try 100.

File: tests/inject/dynamic/style-override.tests.ts [hunk 2]
```diff
@@ -74,4 +74,91 @@ describe('Style override', () => {
...
+        );
+        createOrUpdateDynamicTheme(theme, null, false);
+
+        const style: HTMLStyleElement = document.createElement('style');
+        style.classList.add('testcase-style');
+        container.append(style);
+        style.sheet.insertRule('h1 { color: pink }');
+        style.sheet.insertRule('strong { color: orange }');
+
+        await timeout(1000);
```
Comment (alexanderby) [inline tests/inject/dynamic/style-override.tests.ts:157]: Reduce timeout, because when there will be many tests, they will run slow.

**Review by alexanderby — CHANGES_REQUESTED** [review]

**Comment by alexanderby** [comment]
Why is package-lock changed so much? Didn't you remove it before running `npm install`?

**Comment by Gusted** [comment]
> Why is package-lock changed so much? Didn't you remove it before running npm install?

No?
Let me run it again.

**Comment by Gusted** [comment]
Same result. If I remove the file and then `npm run install` it seems to remove 2 more dependencies.

**Comment by alexanderby** [comment]
What if you copy package-lock from master and run `npm install`? Maybe it will not remove anything?

**Comment by Gusted** [comment]
> 
> 
> What if you copy package-lock from master and run `npm install`? Maybe it will not remove anything?

https://github.com/darkreader/darkreader/pull/4015/commits/c6da916ecec6eafb07e52d455c603776d14e6752
It just added those 3 back, but not the 1K line.

**Comment by alexanderby** [comment]
I think it's my fault, I played with many Karma plugins and seems like didn't run `npm install` when removed them from `package.json`. I will submit a PR.

**Review by alexanderby — APPROVED** [review]

**Comment by alexanderby** [comment]
Great, there's some http -> https switch, I think it's NPM bug.

Approaching my bedtime, I will review some more tomorrow.

**Comment by Gusted** [comment]
> Great, there's some http -> https switch, I think it's NPM bug.

I've got those a lot, doesn't seem to harm anything 🤷🏽 

> Approaching my bedtime, I will review some more tomorrow.

Time to create more PR's =)

**Pull request merged**